smithy-model/src/main/resources/software/amazon/smithy/model/loader/prelude.smithy

```diff
@@ -615,6 +617,33 @@ string pattern
 )
 structure required {}
 
+/// Configures a structure member's resource property mapping behavior.
+@trait(
+    selector: "structure > member",
+    conflicts: [resourceIdentifier],
+    breakingChanges: [{change: "remove"}, {change: "update"}]
+)
+structure property {
+    name: String
+}
+
+/// Explicitly excludes a member from resource property mapping or
+/// enables another trait to carry the same implied meaning.
+@trait(
+    selector: ":is(operation -[input, output]-> structure > member, [trait|trait])",
+    breakingChanges: [{change: "add"}]
+)
+structure notProperty {}
```

Comment:
What if we did something super meta and applied `@notProperty` to itself? We do that already with the `@trait` trait so I know it works. It would just make computing the set of traits that carry `@notProperty` to be a one-step thing rather than computing the set and then adding `@notProperty` manually to it.